Updating packages on start-up

grctest opened the pull request:
This could likely be made redundant by switching to a newer parent docker repo, but it's worthwhile attempting to upgrade especially if new project admins don't delve much into maintaining the apache container via bash.

Related to: https://github.com/marius311/boinc-server-docker/issues/26#issuecomment-330553493

grctest commented:
In fact, we could probably replace the libssl1.0.0 package with openssl instead of installing both.

marius311 commented:
Thanks for looking at this. It looks like libssl1.0.2 does exist in jessie backports (https://packages.debian.org/jessie-backports/libssl1.0.0) I think the correct way to do this is adding the backports repo to sources.list and installing that version outright. In general its not good Dockerfile practice to do an `apt-get upgrade` (https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/#run). I'll try to test it out but I'm a bit swamped so feel free to beat me to it!

grctest commented:
* Updated the php apache parent docker container to version "php:5.6.31-apache".
  * Reason: Lower than 5.6.31 no longer on the php github repo - current no longer maintained?
* Added the backport jessie repo
* Installing libssl1.0.0 and openssl from the backport jessie.

openssl version now shows as:
> OpenSSL 1.0.2l  25 May 2017

No upgrades available when running the 'apt-get upgrade' command, so looks good.

grctest commented:
Regarding the php version upgrade, version 5.6.13 [has several high priority vulnerabilities](https://www.cvedetails.com/vulnerability-list.php?vendor_id=74&product_id=128&version_id=192559&page=1&hasexp=0&opdos=0&opec=0&opov=0&opcsrf=0&opgpriv=0&opsqli=0&opxss=0&opdirt=0&opmemc=0&ophttprs=0&opbyp=0&opfileinc=0&opginf=0&cvssscoremin=0&cvssscoremax=0&year=0&month=0&cweid=0&order=3&trc=40&sha=11e319f39c87a9043fc4a423934a9c6255ca9b7e) where was version 5.6.31 [somehow has none](https://www.cvedetails.com/vulnerability-list/vendor_id-74/product_id-128/version_id-221784/PHP-PHP-5.6.31.html)

marius311 commented:
Made a few tweaks to put stuff into a single RUN command and minimize the number of layers that the image has. Let me know if it looks OK and I'll merge.

grctest commented:
Looks good to me, we don't need to install libssl1.0.0 if we're installing openssl, right? I think it's good to merge.

marius311 commented:
> we don't need to install libssl1.0.0 if we're installing openssl

Exactly, its installed as a dependency. 

Thanks again for this!
